ReactCommon/ReactCommon.podspec

```diff
@@ -32,8 +32,9 @@ Pod::Spec.new do |s|
   s.platforms              = { :ios => "9.0", :tvos => "9.2" }
   s.source                 = source
   s.header_dir             = "ReactCommon" # Use global header_dir for all subspecs for use_framework compatibility
+  s.static_framework       = true
```

Comment:
could you help me understand why this line matters?

Reply:
OK, found the explanation in your summary:

> Each podspec now uses s.static_framework = true. This means that a minimum of CocoaPods 1.5 (released in April 2018) is now required. This is needed so that the __has_include conditions can still work when frameworks are enabled.

I think it's worth exploring not using static_framework, but we can do it outside of this PR. I'm not sure if it's 100% possible given the current directory structure.

Reply:
Just to clear this up completely, here is a more thorough explanation.

In a number of places throughout the codebase, there is code like this:

```
#if RCT_DEV && __has_include(<React/RCTDevLoadingView.h>)
    // Note: RCTDevLoadingView should have been loaded at this point, so no need to allow lazy loading.
    RCTDevLoadingView *loadingView = [weakSelf moduleForName:RCTBridgeModuleNameForClass([RCTDevLoadingView class])
                                       lazilyLoadIfNecessary:NO];
    [loadingView updateProgress:progressData];
#endif
```

The trouble is that this code lives in `React-Core.podspec` but `RCTDevLoadingView` is part of `React-DevSupport`. It works fine when not using frameworks but when `use_frameworks!` is used and `RCT_DEV` is enabled there is an error compiling `React-Core.podspec` as `RCTDevLoadingView` is not found. This is because `React-DevSupport` is not linked with `React-Core`.

The solution is to add `s.static_framework = true` to each of the podspecs. Now all symbols are linked into the same executable so `RCTDevLoadingView` is present at build time. It is necessary to add `s.static_framework = true` to all podspecs as static frameworks are not allowed to have dynamic dependencies.

As mentioned above the main disadvantage of this is that it requires CocoaPods 1.5 or later to be used ([released in April 2018](http://blog.cocoapods.org/CocoaPods-1.5.0/)). Another drawback is that I believe CocoaPods complains if a target has transitive static dependencies. This means any third party podspec that depends on `React` would also need to have `s.static_framework = true`.

> I think it's worth exploring not using static_framework, but we can do it outside of this PR. I'm not sure if it's 100% possible given the current directory structure.

The only way I can think of that would be possible would be to make `React-DevSupport` a subspec of `React-Core` as it was before the podspec split.

Reply:
> This means any third party podspec that depends on React would also need to have s.static_framework = true.

Yeah, this is my worry: it could be too limiting for those libs.


> The only way I can think of that would be possible would be to make React-DevSupport a subspec of React-Core as it was before the podspec split.

I was actually trying this on top of your PR, but got a dependency cycle error, so I didn’t proceed. Perhaps it’s worth another look?

Reply:
> I was actually trying this on top of your PR, but got a dependency cycle error, so I didn’t proceed. Perhaps it’s worth another look?

I can take a look.

Reply:
Ah, I remember the problem with that now. `DevSupport` depends on `React-RCTWebSocket` which depends on `React-Core`.

If we make `React-RCTWebSocket` a subspec too, it will work. I'll open a PR.

Reply:
Okay, I've done that here: https://github.com/facebook/react-native/pull/25816